tasks/eth/021-holocene-protocol-versions/SignFromJson.s.sol

```diff
@@ -0,0 +1,89 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.15;
+
+import {SignFromJson as OriginalSignFromJson} from "script/SignFromJson.s.sol";
+import {Simulation} from "@base-contracts/script/universal/Simulation.sol";
+import {Proxy} from "@eth-optimism-bedrock/src/universal/Proxy.sol";
+import {SystemConfig} from "@eth-optimism-bedrock/src/L1/SystemConfig.sol";
+import {ProtocolVersions, ProtocolVersion} from "@eth-optimism-bedrock/src/L1/ProtocolVersions.sol";
+import {Types} from "@eth-optimism-bedrock/scripts/Types.sol";
+import {console2 as console} from "forge-std/console2.sol";
+import {stdToml} from "forge-std/StdToml.sol";
+import {Vm, VmSafe} from "forge-std/Vm.sol";
+import {LibString} from "solady/utils/LibString.sol";
+
+contract SignFromJson is OriginalSignFromJson {
+    using LibString for string;
+
+    // Chains for this task.
+    string constant l1ChainName = "mainnet";
+    string constant l2ChainName = "op";
+
+    uint256 constant protoVerHolocene = 0x0000000000000000000000000000000000000009000000000000000000000000;
```

Comment:
for a reusable check module, ideally this would be created from `major, minor, path, prerelease` `uint64` constructor arguments.

Reply:
Yes although I think they should be uint32 https://specs.optimism.io/protocol/superchain-upgrades.html?highlight=protocolversions#protocol-version-format